packages/clarity-js/src/data/metadata.ts

```diff
@@ -19,19 +19,29 @@ export function start(): void {
   data = {
     projectId: config.projectId || hash(location.host),
     userId: user(),
-    sessionId: s.id,
+    sessionId: s.session,
     pageNum: s.count
   }
 
+  // For backward compatibility; remove in future iterations (v0.6.11)
+  if (config.upload && typeof config.upload === Constant.String && (config.upload as string).indexOf(Constant.HTTPS) === 0) {
+    let url = config.upload as string;
+    config.server = url.substr(0, url.indexOf("/", Constant.HTTPS.length));
+    config.upload = config.server.length > 0 && config.server.length < url.length ? url.substr(config.server.length + 1) : url;
```

Comment:
>&& [](start = 45, length = 2)

All this string munging is hard to follow. What are you doing here? Can you add a comment about what you this is for?

Reply:
I'll add comments to explain it more. I am hoping we can remove this bit with the next version.

Reply:
Done